nixos/modules/services/autotierfs.nix

```diff
@@ -0,0 +1,87 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.autotierfs;
+  ini = pkgs.formats.ini { };
+  format = lib.types.attrsOf ini.type;
+
+  configName = builtins.replaceStrings [ "/" ] [ "_" ];
+  configFiles = builtins.mapAttrs (
+    name: val: ini.generate "${configName name}-autotier.conf" val
+  ) cfg.settings;
+
+  getMountDeps = settings: builtins.catAttrs "Path" (builtins.attrValues settings);
+
+  mountPaths = builtins.attrNames cfg.settings;
+in
+{
+  options.services.autotierfs = {
+    enable = lib.mkEnableOption "the autotier passthrough tiering filesystem";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.autotier;
+      defaultText = lib.options.literalExpression "pkgs.autotier";
+      description = "Configured package for the filesystem and its cli.";
+    };
+    settings = lib.mkOption {
+      type = lib.types.nullOr format;
+      default = null;
+      description = ''
+        The contents of the configuration file for autotier.
+        Config can be found in the [autotier repo](https://github.com/45Drives/autotier#configuration)
+      '';
+      example = {
+        "/mnt/autotier" = {
+          Global = {
+            "Log Level" = 1;
+            "Tier Period" = 1000;
+            "Copy Buffer Size" = "1 MiB";
+          };
+          "Tier 1" = {
+            Path = "/mnt/tier1";
+            Quota = "30GiB";
+          };
+          "Tier 2" = {
+            Path = "/mnt/tier2";
+            Quota = "200GiB";
+          };
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.enable -> cfg.settings != null;
+        message = "Autotier needs a config file to know how to tier your paths.";
+      }
+    ];
+
+    system.fsPackages = [ cfg.package ];
+
+    # Not necessary for module to work but makes it easier to pass config into cli
+    environment.etc = builtins.mapAttrs (n: v: { source = v; }) (
+      lib.attrsets.mapAttrs' (
+        name: value: lib.attrsets.nameValuePair "${(configName name)}.conf" (value)
+      ) configFiles
+    );
```

Comment:
I'm not sure about this part. Without it, getting the config for using the cli is a PITA, but idk how to format the name of the config files to be more human friendly. In the attrset they're just ``"path" = "config"`` but ofc you cant do that with a raw file bc it actually ends up making it as ``/mnt/location`` with ``location = config``

Reply:
I think generating a value like this is okay.